streaming/base/stream.py

```diff
@@ -145,6 +149,15 @@ def __init__(self,
             self.keep_zip = keep_zip
             self.safe_keep_zip = self.keep_zip or self.remote in {None, self.local}
 
+    def _get_temporary_directory(self) -> str:
+        root = tempfile.gettempdir()
```

Comment:
What would be the value of `root` when running on 2 GPUs? Does each rank gives the same `root` directory ?

Reply:
Yep they use the same `root`. I've changed the tests to use 2 GPUs and they all pass, but I'm getting a weird error at the end of the pytest saying it didn't exit correctly

```
========================= 4 passed, 1 warning in 1.17s =========================

----------End global rank 1 STDOUT----------
----------Begin global rank 1 STDERR----------

----------End global rank 1 STDERR----------
ERROR:__main__:Global rank 0 (PID 6785) exited with code -15
```